internal/rootcoord/create_collection_task.go

```diff
@@ -66,18 +66,15 @@ func (t *createCollectionTask) validate() error {
 		return err
 	}
 
+	// 1. check shard number
 	shardsNum := t.Req.GetShardsNum()
 
 	cfgMaxShardNum := Params.RootCoordCfg.DmlChannelNum.GetAsInt32()
 	if shardsNum > cfgMaxShardNum {
 		return fmt.Errorf("shard num (%d) exceeds max configuration (%d)", shardsNum, cfgMaxShardNum)
 	}
 
-	cfgShardLimit := Params.ProxyCfg.MaxShardNum.GetAsInt32()
-	if shardsNum > cfgShardLimit {
-		return fmt.Errorf("shard num (%d) exceeds system limit (%d)", shardsNum, cfgShardLimit)
```

Comment:
I believe it is still necessary to maintain this logic. It restricts the number of shards for a single table from exceeding the maximum configured shard count.